### PR TITLE
refactor GAMESS-US easyblock to directly write install.info (v2)

### PR DESCRIPTION
**Pull request by lexming**
Follow-up of https://github.com/easybuilders/easybuild-easyblocks/pull/2791

Main changes:
* remove QA, instead configure installation by directly writing the `install.info` file
* update configuration options and supported dependencies from version 2020xxxx to 2023xxxx
* run all tests for non-MPI builds in serial
* run tests for MPI builds in parallel with the same parallelization of the install and exclude those tests known to only work in serial

Fixes https://github.com/easybuilders/easybuild-easyconfigs/issues/19791 (commit https://github.com/easybuilders/easybuild-easyblocks/pull/3047/commits/1bfd25373dca72660f725b65e87833420d8dd936)

### CODE REVIEW

**Comment by lexming** [comment]
Test reports:
* v20230930-R2
    * MPI target (`ddi_comm=mpi`): GNU compilers + OpenBLAS and OpenMPI 
        * Intel broadwell (AVX2): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1860915239
        * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1860796615
    * Sockets target (`ddi_comm=socket`): Intel compilers + MKL and OpenMP
        * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19452#issuecomment-1865928668 
* v20220930-R2
    * MPI target (`ddi_comm=mpi`): GNU compilers + OpenBLAS and OpenMPI 
        * Intel broadwell (AVX2): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1860915239
        * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1860796615

File: easybuild/easyblocks/g/gamess_us.py [hunk 1]
```diff
@@ -33,24 +33,35 @@
...
-from easybuild.tools.run import run_cmd, run_cmd_qa
-from easybuild.tools.systemtools import get_platform_name
-from easybuild.tools import toolchain
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import POWER, X86_64
+from easybuild.tools.systemtools import get_cpu_architecture
+from easybuild.tools import LooseVersion, toolchain
+
+GAMESS_INSTALL_INFO = 'install.info'
+GAMESS_MPI_TEST_BLACKLIST = [
```
Comment (boegel) [inline easybuild/easyblocks/g/gamess_us.py:57]: maybe rename this to `GAMESS_SERIAL_TESTS` ?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/g/gamess_us.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/g/gamess_us.py]: sure, fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/3047/commits/c79e6e250d03ef6013b7d63677bfd25dd8f65b83 alongside a bit of refactor to run all tests in serial for non-mpi builds

**Comment by lexming** [comment]
Following the discussion in our last conf call I updated the code executing the tests to run all tests in serial for non-MPI builds of GAMESS-US. No further changes for the tests of MPI builds, those must be run in parallel (`rungms` does not allow to run in a single core in mpi mode) and we filter out those tests known to not be compatible with a parallel run.

**Comment by lexming** [comment]
### Tests Reports

I made 2 configurations complementing each other to cover as many options as possible with the least amount of tests:
1. MPI build with _foss_ stack
2. Non-MPI build with _intel_ stack

### v20230930-R2

* MPI target (`ddi_comm=mpi`): GNU compilers + OpenBLAS and OpenMPI 
    * Intel broadwell (AVX2): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1925410576
    * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1925410609
* Sockets target (`ddi_comm=socket`): Intel compilers + MKL and OpenMP
    * Intel broadwell (AVX2): https://github.com/easybuilders/easybuild-easyconfigs/pull/19452#issuecomment-1925397544
    * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19452#issuecomment-1925397099 

### v20220930-R2

* MPI target (`ddi_comm=mpi`): GNU compilers + OpenBLAS and OpenMPI 
    * Intel broadwell (AVX2): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1925410576
    * Intel skylake (AVX512): https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-1925410609

**Comment by lexming** [comment]
## Latest Test reports

### v20230930-R2

* MPI target (ddi_comm=mpi): GNU compilers + OpenBLAS and OpenMPI 
    * https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-2100044752

### v20220930-R2

* MPI target (ddi_comm=mpi): GNU compilers + OpenBLAS and OpenMPI 
    * https://github.com/easybuilders/easybuild-easyconfigs/pull/19310#issuecomment-2100044752

**Comment by lexming** [comment]
Fixes https://github.com/easybuilders/easybuild-easyconfigs/issues/19791 (commit https://github.com/easybuilders/easybuild-easyblocks/pull/3047/commits/1bfd25373dca72660f725b65e87833420d8dd936)

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @lexming!

**Pull request merged**